### PR TITLE
tests for gurobi solution pool

### PR DESCRIPTION
**Pull request by michaelbynum**
## Summary/Motivation:
This PR fixes bugs in the Appsi interface to Gurobi when making use of the solution pool.

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/contrib/appsi/solvers/tests/test_gurobi_persistent.py [hunk 1]
```diff
@@ -301,6 +392,21 @@ def test_nonconvex2(self):
...
+        opt.gurobi_options['PoolSearchMode'] = 2
+        res = opt.solve(m)
+        num_solutions = opt.get_model_attr('SolCount')
+        self.assertEqual(num_solutions, 3)
+        res.solution_loader.load_vars(solution_number=0)
+        self.assertAlmostEqual(pe.value(m.obj.expr), 6.431184939357673)
+        res.solution_loader.load_vars(solution_number=1)
+        self.assertAlmostEqual(pe.value(m.obj.expr), 6.584793218502477)
+        res.solution_loader.load_vars(solution_number=2)
+        self.assertAlmostEqual(pe.value(m.obj.expr), 6.592304628123309)
```
Comment (jsiirola) [inline pyomo/contrib/appsi/solvers/tests/test_gurobi_persistent.py:408]: How fragile will these tests be with different versions of Gurobi?  Is it sufficient to just assert that the objective values are increasing?

Comment (michaelbynum) [reply pyomo/contrib/appsi/solvers/tests/test_gurobi_persistent.py:408]: I don't think these should be fragile. Setting the `PoolSearchMode` to `2` and `PoolSolutions` to 3 instructs Gurobi to find the best 3 solutions. These are not solutions Gurobi happened to come across during the solve. They are actually the best 3.

Comment (jsiirola) [reply pyomo/contrib/appsi/solvers/tests/test_gurobi_persistent.py:408]: Sounds good.

**Review by jsiirola — APPROVED** [review]
Looks good, pending successful tests.

**Comment by whart222** [comment]
The PR looks good to me as well.  (Says Bill, waiting for the tests to finish...)

**Pull request merged**